.github/workflows/pre-commit.yml

```diff
@@ -11,6 +11,15 @@ jobs:
     steps:
       - uses: actions/checkout@v2
       - uses: actions/setup-python@v2
+      - name: Cache multiple paths
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cache/pip
+            /opt/hostedtoolcache/Python/
+            ~\AppData\Local\pip\Cache
+            c:\hostedtoolcache\windows\python\
+          key: ${{ runner.os }}-build-${{ env.cache-name }}
       # This is required since pylint is used in pre-commit as a local hook
       - name: Install dependencies
```

Comment:
Given all dependencies are taken care by pre-commit now, we can remove the `Install dependencies` step that follows the `caching step`. So in this workflow also we would need to cache only `~/.cache/pre-commit` since the only useful action is `pre-commit/action@v2.0.0`

Reply:
Also this cache key will collide with the test caches.

Reply:
I have removed the caching action from the `pre-commit` GH action

Reply:
Are you saying the `actions/setup-python@v2` takes care of caching (the thing we are trying to do)?

Reply:
@s-weigand what do you suggest we do to keep the keys unique to the different processes? Add a new identifier that can help us separate them apart?

Reply:
> ~/.cache/pre-commit

@girip11 I'm removing all folders except `~/.cache/pre-commit` in `pre-commit.yml` as suggested by you

Reply:
@neomatrix369 you can also remove the whole "install dependencies" step from this particular workflow (I should've done that in a different PR, but we might as well do it now)

Reply:
So we can remove the caching step from this workflow along with the `Install dependencies step`.
So the final steps would look like

```YAML
    steps:
      - uses: actions/checkout@v2
      - uses: actions/setup-python@v2
      - uses: pre-commit/action@v2.0.0
```

If we want to be safe from the tool breakages due to new releases of python (pylint broken on python 3.9), we can fix the python version during the `setup-python` step
```YAML
    steps:
      - uses: actions/checkout@v2
      - uses: actions/setup-python@v2
        with:
          python-version: "3.8"
      - uses: pre-commit/action@v2.0.0
```